jlexer/lexer_test.go

```diff
@@ -312,3 +312,22 @@ func TestJsonNumber(t *testing.T) {
 		}
 	}
 }
+
+func TestFetchStringUnterminatedString(t *testing.T) {
+	for _, test := range []struct {
+		data []byte
+	}{
+		{data: []byte(`"sting without trailing literal`)},
+		{data: []byte(`"\"`)},
+		{data: []byte{'"'}},
+	} {
+		l := Lexer{Data: test.data}
+		l.fetchString()
+		if l.pos > len(l.Data) {
+			t.Errorf("pos should not be greater than length of Data")
```

Comment:
Please indicate what test case failed, e.g.:

t.Errorf("fetchString(%s): pos should not be greater than length of Data", test.data) 

Same goes for the error below.